index.bs

```diff
@@ -2921,35 +2920,43 @@ the same operation or constructor.
         the argument on which the ellipsis appears counts as a single argument.
 
         Note: So <code>void f(long x, long... y);</code> is considered to be declared to take two arguments.
-
-    1.  Let |m| be the maximum of |maxarg| and |N|.
+    1.  Let |max| be [=max=](|maxarg|, |N|).
     1.  For each operation, extended attribute or callback function |X| in |F|:
-        1.  Let |n| be the number of arguments |X| is declared to take.
-        1.  Let |t|<sub>0..|n|−1</sub> be a list of types, where |t|<sub>|i|</sub>
-            is the type of |X|’s argument at index |i|.
-        1.  Let |o|<sub>0..|n|−1</sub> be a list of [=optionality values=], where |o|<sub>|i|</sub>
-            is “variadic” if |X|’s argument at index |i| is a final, variadic argument,
-            “optional” if the argument is [=optional argument|optional=],
-            and “required” otherwise.
-        1.  Add to |S| the tuple &lt;|X|, |t|<sub>0..|n|−1</sub>, |o|<sub>0..|n|−1</sub>&gt;.
+        1.  Let |arguments| be the [=list=] of arguments |X| is declared to take.
+        1.  Let |n| be the [=list/size=] of |arguments|.
+        1.  Let |types| be a [=list=] of types.
+        1.  Let |optionality_values| be a list of [=optionality values=].
+        1.  [=list/For each=] |argument| in |arguments|:
+            1.  [=list/Append=] the type of |argument| to |types|.
+            1.  [=list/Append=] "variadic" to |optionality_values| if |argument| is a final, variadic argument,
+                "optional" if |argument| is [=optional argument|optional=],
+                and "required" otherwise.
+        1.  [=set/Append=] the [=tuple=] (|X|, |types|, |optionality_values|) to |S|.
         1.  If |X| is declared to be [=variadic=], then:
-            1.  For every integer |i|, such that |n| ≤ |i| ≤ |m|−1:
-                1.  Let |u|<sub>0..|i|</sub> be a list of types,
-                    where |u|<sub>j</sub> = |t|<sub>j</sub> (for |j| &lt; |n|) and
-                    |u|<sub>j</sub> = |t|<sub>|n|−1</sub> (for |j| ≥ |n|).
-                1.  Let |p|<sub>0..|i|</sub> be a list of [=optionality values=],
-                    where |p|<sub>j</sub> = |o|<sub>j</sub> (for |j| &lt; |n|) and
-                    |p|<sub>j</sub> = “variadic” (for |j| ≥ |n|).
-                1.  Add to |S| the tuple &lt;|X|, |u|<sub>0..|i|</sub>, |p|<sub>0..|i|</sub>&gt;.
-        1.  Initialize |i| to |n|−1.
-        1.  While |i| ≥ 0:
-            1.  If argument |i| of |X| is not
-                [=optional argument|optional=]
-                (i.e., it is not marked as <code>optional</code> and is not
-                a final, variadic argument), then break this loop.
-            1.  Otherwise, add to |S| the tuple &lt;|X|, |t|<sub>0..|i|−1</sub>, |o|<sub>0..|i|−1</sub>&gt;.
+            1.  [=list/For each=] |i| in [=the range=] |n| to |max|−1, inclusive:
```

Comment:
I think we usually use spaces on either side of the minus sign? Also, it's worth confirming this is a minus sign − instead of a dash.

Here and elsewhere.